esp/logging/logginglib/modularlogagent.cpp

```diff
@@ -0,0 +1,1526 @@
+                    return createNewFile(file, pattern, systemVariables);
+                }
+                else
+                {
+                    // The filename is as unique as it can be, as configured. Add "random" text to
+                    // resolve the conflict.
+                    StringBuffer path, prefix, extension;
+                    splitFilename(filePath, nullptr, &path, &prefix, &extension);
+                    newIO.setown(createUniqueFile(path, prefix, extension, filePath.clear()));
+                }
+            }
+            else
+            {
+                newIO.setown(newFile->openShared(IFOcreate, IFSHnone));
+            }
+            if (!newIO)
+            {
+                ierrlog(Major, "failed to create and open new file '%s'", filePath.str());
+            }
+            else if (!m_header.isEmpty() && newIO->write(0, m_header.length(), m_header) != m_header.length())
+            {
+                ierrlog(Major, "failed to add header to '%s'", filePath.str());
+            }
+            else
+            {
+                file.m_io.setown(newIO.getClear());
+                file.m_filePath.set(filePath);
+                file.m_lastWrite.setNow();
+                result = true;
+            }
+        }
+        else
+        {
+            ierrlog(Major, "failed to create directory structure for '%s'", filePath.str());
+        }
+    }
+    catch(IException* e)
+    {
+        StringBuffer msg;
+        ierrlog(Major, "exception creating and opening new file '%s': %s", filePath.str(), e->errorMessage(msg).str());
+        e->Release();
+    }
+    return result;
+}
+
+/**
+ * Appends a section of data to a targeted file, capturing and recording errors.
+ */
+bool CFileTarget::writeChunk(File& file, offset_t pos, size32_t len, const char* data) const
+{
+    bool result = false;
+    try
+    {
+        size32_t written = file.m_io->write(file.m_io->size(), len, data + pos);
+        if (written != len)
+            ierrlog(Major, "failed to write %u bytes to '%s'; wrote %u", len, file.m_filePath.str(), written);
+        else
+            result = true;
+        file.m_lastWrite.setNow();
+    }
+    catch(IException* e)
+    {
+        StringBuffer msg;
+        ierrlog(Major, "exception writing %u bytes to '%s': %s", len, file.m_filePath.str(), e->errorMessage(msg).str());
+        e->Release();
+    }
+    return result;
+}
+
+//////////////////// CEspLogAgent ////////////////////
+
+const char* CEspLogAgent::getName()
+{
+    return m_name;
+}
+
+bool CEspLogAgent::init(const char* name, const char* type, IPTree* configuration, const char* process)
+{
+    if (!configuration)
+        throw MakeStringException(-1, "%s[%s]: missing configuration", ModularLogAgent::moduleAgent, name);
+
+    m_name.set(name);
+
+    Linked<IPTree> effectiveConfiguration;
+    const char* configFile = configuration->queryProp(propConfiguration);
+    if (!isEmptyString(configFile))
```

Comment:
So this clause of the if statement handles split configuration and the else handles inline?

Reply:
A non-empty value for propConfiguration does imply the intent to use a split configuration, while an empty value can only mean an inline configuration.